core/vm/interpreter.go

```diff
@@ -124,6 +124,9 @@ func (in *EVMInterpreter) Run(contract *Contract, input []byte, readOnly bool) (
 	// Make sure the readOnly is only set if we aren't in readOnly yet.
 	// This also makes sure that the readOnly flag isn't removed for child calls.
 	if readOnly && !in.readOnly {
+		if !in.evm.chainRules.IsByzantium {
+			panic("readOnly execution requested on wrong chainRules");
```

Comment:
This is a new addition here, which I think may even make sense without the rest of the changes. Not sure about how well `panic` is perceived, but seen it used occasionally in similar context.

Reply:
We can leave this here if we want to do a test-run, but I think we should remove it from production code. 
I'd also be fine with not having the panic here at all.

Reply:
Is there any option besides panic? I agree that panic is kind of risky as it may result in inconsistent shutdowns, but if this is hit that is a potential consensus issue.

Reply:
We sometimes use panic, when something has failed beyond where we can do anything about it. 
We don't usually actively look for failures to panic on, unless it's a piece of new code which we want to test during some benchmark runs. 

So for prod code, we'll remove this. I'm personally pretty fine with believing that this will never happen, but if you want it for test-runs, then sure, we can keep it for a while

Reply:
Should we keep this panic for running a sync or just drop it now? I am fine either way. Should a comment be kept once the code is dropped?

Reply:
@holiman please give some advice on this.

Reply:
IMO drop it

Reply:
Dropped.